crates/modelardb_server/src/main.rs

```diff
@@ -110,9 +113,16 @@ fn main() -> Result<(), String> {
         Runtime::new().map_err(|error| format!("Unable to create a Tokio Runtime: {error}"))?,
     );
 
-    // Check if a remote data folder was provided and can be accessed. These checks are performed
+    // If a remote data folder was provided, check that it can be accessed. This check is performed
     // after parse_command_line_arguments() as the Tokio Runtime is required.
-    check_remote_data_folder(&runtime, &data_folders.remote_data_folder)?;
+    if let Some(remote_data_folder) = &data_folders.remote_data_folder {
+        // unwrap() is safe since if there is a remote data folder, there is always a third argument.
+        let remote_data_folder_type = arguments.get(2).unwrap().split_once("://").unwrap().0;
```

Comment:
To only parse the command line argument in `argument_to_remote_object_store()` and restrict `remote_data_folder_type` to the supported object stores, maybe create a enum with a variant for each object store which contain `Arc<dyn ObjectStore>`. Then after `validate_remote_data_folder()` has been run the object store can be extracted from the enum as we no longer need to know its type. Alternative a enum without any data could be created and returned from `argument_to_remote_object_store()` in a pair together with `Arc<dyn ObjectStore>`. Using either methods would remove the `&str` parameter from `validate_remote_data_folder()` which only accepts a very small number of arguments and remove the need for parsing `argument.get(2)` in two different places.

Reply:
I have changed the `validate_remote_data_folder()` function to accept an enum instead. However, i did not find a clean way to extract the enum value in `argument_to_remote_object_store` and pass it back. 

It would either require passing a runtime into the function to validate right after creating the object store or require that we pass the value back to main as you mention. Passing a runtime down breaks some tests so that is not ideal and passing the value back makes the code very messy. I did not find a way to cleanly extract the object store value from the enum when the enum variant is not known and passing back a pair only to avoid having to read the third argument twice seems overkill.

Reply:
I agree.